cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexUpdaterTest.java

```diff
@@ -1041,6 +1041,114 @@ public Void call(SQLDatabase db) throws Exception {
+            // Assert that the index made is still there
+            Set<Map.Entry<String, Object>> entries = im.listIndexes().entrySet();
+            Assert.assertEquals("There should be 1 index.", 1, entries.size());
+            Map.Entry<String, Object> actualIndex = entries.iterator().next();
+            Assert.assertEquals("There should be the named index.", "testIndex", actualIndex
+                    .getKey());
+            Map<String, Object> actualIndexMetadata = (Map<String, Object>) actualIndex.getValue();
+            Assert.assertEquals("The name property of the index should be correct.", "testIndex",
+                    actualIndexMetadata.get("name"));
+            long actualSequence = getIndexSequenceNumber("testIndex");
+            Assert.assertEquals("The sequence should still be 1 for 1 document created.", 1,
+                    actualSequence);
+        } finally {
+            im.close();
+        }
+    }
+
+    @Test
+    public void indexUpdatesPersistExistingIndex() throws Exception {
+        long exepctedSequence = 0l;
+        try {
+            DocumentRevision rev = new DocumentRevision("mike12");
+            // body content: { "name" : "mike",  "age" : 12, "pet" : "cat" }
+            Map<String, Object> bodyMap = new HashMap<String, Object>();
+            bodyMap.put("name", "mike");
+            bodyMap.put("age", 12);
+            bodyMap.put("pet", "cat");
+            rev.setBody(DocumentBodyFactory.create(bodyMap));
+            ds.createDocumentFromRevision(rev);
+
+            createIndex("testIndex", Collections.singletonList((Object) "name"));
+            exepctedSequence = getIndexSequenceNumber("testIndex");
+            Assert.assertEquals("The expected sequence should be 1 for 1 document created.", 1,
+                    exepctedSequence);
+        } finally {
+            im.close();
+        }
+
+        // Now get a new IndexManager and queue and update the test fields accordingly
+        im = new IndexManager(ds);
+        indexManagerDatabaseQueue = TestUtils.getDBQueue(im);
+
+        try {
+            // Create another document to update the DB
+            DocumentRevision rev = new DocumentRevision("mike23");
+            // body content: { "name" : "mike",  "age" : 23, "pet" : "parrot" }
+            Map<String, Object> bodyMap = new HashMap<String, Object>();
+            bodyMap.put("name", "mike");
+            bodyMap.put("age", 23);
+            bodyMap.put("pet", "cat");
+            rev.setBody(DocumentBodyFactory.create(bodyMap));
+            ds.createDocumentFromRevision(rev);
+
+            im.updateAllIndexes();
+            exepctedSequence = getIndexSequenceNumber("testIndex");
+            Assert.assertEquals("The expected sequence should be 2 for 2 documents created.", 2,
+                    exepctedSequence);
+        } finally {
+            im.close();
+        }
+
+        // Now get a new IndexManager and queue and update the test fields accordingly
```

Comment:
Replaced with the hopefully more clear 
`// Get a new IndexManager instance and extract its queue`
in 529262a